src/components/fields/location-field/location-helper.ts

```diff
@@ -156,6 +160,22 @@ export namespace LocationHelper {
 	// =========================================================================
 	export const reverseGeocode = async ({ options, ...others }: TReverseGeocodeParams): Promise<IResultsMetaData> => {
 		const locationList = await mapService.reverseGeocode(others);
```

Comment:
can we have the `isCoordinateInBounds()` check here instead?

because this function is also manually filtering `JOHOR (MALAYSIA)` and supports an `excludeNonSG` option, we can have it throw an error instead.

in addition, `reverseGeocode()` is used in other places but don't have the same boundary check, so it should have that check in here to maintain the same behaviour.